### PR TITLE
Turn some variables and functions into static

### PR DESCRIPTION
**Pull request by cyyever**
Re-check some files and mark variables and functions into static and fix other warnings.

### CODE REVIEW

**Review by eqy — COMMENTED** [review]
Would we expect to get a reliable signal from CI for this, or would it potentially break downstream extensions that expect to use some `C10_API` definitions?

**Comment by cyyever** [comment]
> Would we expect to get a reliable signal from CI for this, or would it potentially break downstream extensions that expect to use some `C10_API` definitions?

The changed C10_API variables/functions have no exposed declarations so that I'm not sure. But they seems to be used internally.

File: torch/csrc/Device.cpp [hunk 1]
```diff
@@ -205,6 +213,7 @@ PyObject* THPDevice_call(PyObject* self, PyObject* args, PyObject* kwargs) {
       .ptr();
   END_HANDLE_TH_ERRORS
 }
+*/
```
Comment (ezyang) [inline torch/csrc/Device.cpp:216]: ?

Comment (cyyever) [reply torch/csrc/Device.cpp:216]: Rebase conflicts.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**